docs-website/topics/serialization-json-configuration.md

```diff
@@ -0,0 +1,707 @@
+
+```kotlin
+// Imports declarations from the serialization and JSON handling libraries
+import kotlinx.serialization.*
+import kotlinx.serialization.json.*
+
+//sampleStart
+// Configures a Json instance to allow special floating-point values
+val format = Json { allowSpecialFloatingPointValues = true }
+
+@Serializable
+class Data(
+    val value: Double
+)
+
+fun main() {
+    val data = Data(Double.NaN)
+    // Produces a non-standard JSON output used for representing special floating-point values
+    println(format.encodeToString(data))
+    // {"value":NaN}
+}
+//sampleEnd
+```
+{kotlin-runnable="true"}
+
+### Specify class discriminator for polymorphism
+
+When working with [polymorphic data](serialization-polymorphism.md), you can use the
+[`classDiscriminator`](https://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-json/kotlinx.serialization.json/-json-builder/class-discriminator.html) property to specify a key name that identifies the type of the serialized polymorphic object.
+Combined with an [explicit serial name defined with the `@SerialName` annotation](serialization-customization-options.md#customize-serial-names),
+this approach gives you full control over the resulting JSON structure:
+
+```kotlin
+// Imports declarations from the serialization and JSON handling libraries
+import kotlinx.serialization.*
+import kotlinx.serialization.json.*
+
+//sampleStart
+// Configures a Json instance to use a custom class discriminator
+val format = Json { classDiscriminator = "#class" }
+
+@Serializable
+sealed class Project {
+    abstract val name: String
+}
+
+// Specifies a custom serial name for the OwnedProject class
+@Serializable
+@SerialName("owned")
+class OwnedProject(override val name: String, val owner: String) : Project()
+
+// Specifies a custom serial name for the SimpleProject class
+@Serializable
+@SerialName("simple")
+class SimpleProject(override val name: String) : Project()
+
+fun main() {
+    val simpleProject: Project = SimpleProject("kotlinx.serialization")
+    val ownedProject: Project = OwnedProject("kotlinx.coroutines", "kotlin")
+
+    // Serializes SimpleProject with #class: "simple"
+    println(format.encodeToString(simpleProject))
+    // {"#class":"simple","name":"kotlinx.serialization"}
+
+    // Serializes OwnedProject with #class: "owned"
+    println(format.encodeToString(ownedProject))
+    // {"#class":"owned","name":"kotlinx.coroutines","owner":"kotlin"}
+}
+//sampleEnd
+```
+{kotlin-runnable="true"}
+
+> [`@JsonClassDiscriminator`](https://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-json/kotlinx.serialization.json/-json-class-discriminator/) is [Experimental](components-stability.md#stability-levels-explained). To opt in, use the `@OptIn(ExperimentalSerializationApi::class)` annotation or the compiler option `-opt-in=kotlinx.serialization.ExperimentalSerializationApi`.
```

Comment:
Experimentality warning about `JsonClassDiscriminator` before introducing `JsonClassDiscriminator` itself is strange. This block should be moved one paragraph down.